internal/cli/streams/privatelink/create_test.go

```diff
@@ -0,0 +1,149 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package privatelink
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/mocks"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20241113004/admin"
+)
+
+const fileName = "test-privateLink.json"
+
+func TestCreateOpts_Run(t *testing.T) {
+	validPrivateLinkConfigFileContents := `
+		{
+			"provider": "Azure",
+			"region": "US_EAST_2",
+			"serviceEndpointId": "/subscriptions/fd01adff-b37e-4693-8497-83ecf183a145/resourceGroups/test-rg/providers/Microsoft.EventHub/namespaces/test-namespace",
+			"dnsDomain": "test-namespace.servicebus.windows.net"
+		}
+	`
+
+	testCases := []struct {
+		name         string
+		fileContents string
+		wantErr      require.ErrorAssertionFunc
+	}{
+		{
+			name:         "no file passed in",
+			fileContents: "",
+			wantErr:      require.Error,
+		},
+		{
+			name: "file does not contain a provider",
+			fileContents: `
+				{
+					"region": "US_EAST_2",
+					"serviceEndpointId": "/subscriptions/fd01adff-b37e-4693-8497-83ecf183a145/resourceGroups/test-rg/providers/Microsoft.EventHub/namespaces/test-namespace",
+					"dnsDomain": "test-namespace.servicebus.windows.net"
+				}
+			`,
+			wantErr: func(tt require.TestingT, err error, _ ...any) {
+				require.ErrorContains(tt, err, "provider missing")
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			fs := afero.NewMemMapFs()
+
+			if len(tc.fileContents) > 0 {
+				require.NoError(t, afero.WriteFile(fs, fileName, []byte(tc.fileContents), 0600))
+			}
+
+			createOpts := &CreateOpts{
+				fs:       fs,
+				filename: fileName,
+			}
+
+			tc.wantErr(t, createOpts.Run())
+		})
+	}
+
+	t.Run("should call the store create privateLink method with the correct parameters", func(t *testing.T) {
+		fs := afero.NewMemMapFs()
+		require.NoError(t, afero.WriteFile(fs, fileName, []byte(validPrivateLinkConfigFileContents), 0600))
+
+		ctrl := gomock.NewController(t)
+		mockStore := mocks.NewMockPrivateLinkCreator(ctrl)
+
+		createOpts := &CreateOpts{
+			store:    mockStore,
+			fs:       fs,
+			filename: fileName,
+		}
+
+		expectedProvider := "Azure"
+		expectedRegion := "US_EAST_2"
+		expectedServiceEndpointID := "/subscriptions/fd01adff-b37e-4693-8497-83ecf183a145/resourceGroups/test-rg/providers/Microsoft.EventHub/namespaces/test-namespace"
+		expectedDNSDomain := "test-namespace.servicebus.windows.net"
+
+		expected := atlasv2.NewStreamsPrivateLinkConnection()
+		expected.Provider = &expectedProvider
+		expected.Region = &expectedRegion
+		expected.ServiceEndpointId = &expectedServiceEndpointID
+		expected.DnsDomain = &expectedDNSDomain
```

Comment:
you never read the "expected values so just use setters to avoid pointer logic 
```suggestion
	expected := atlasv2.NewStreamsPrivateLinkConnection()
	expected.SetProvider("Azure")
	expected.SetRegion("US_EAST_2")
	expected.SetServiceEndpointId("/subscriptions/fd01adff-b37e-4693-8497-83ecf183a145/resourceGroups/test-rg/providers/Microsoft.EventHub/namespaces/test-namespace")
	expected.SetDnsDomain("test-namespace.servicebus.windows.net")
```